stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -232,6 +232,15 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 %  end
   }
 
+  /// Creates an empty instance.
+  public init() {
+    // This is known to create a suitably-aligned pointer on all
+    // currently-supported platforms.
```

Comment:
I would like to reserve the right to consider any pointer in the zero page invalid. `Pointer.swift` already sets a precedent for producing a dummy pointer:
```
    let lastAlignedValue = ~(MemoryLayout<FromElement>.alignment - 1)
    let lastAlignedPointer = UnsafeRawPointer(bitPattern: lastAlignedValue)!
```